pkg/trantor/appmodule.go

```diff
@@ -24,121 +28,80 @@ type AppModule struct {
-	protocolModule t.ModuleID
 }
 
 // NewAppModule creates a new AppModule.
-func NewAppModule(appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) *AppModule {
-	return &AppModule{
-		appLogic:       appLogic,
-		transport:      transport,
-		protocolModule: protocolModule,
-	}
-}
+func NewAppModule(appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) modules.PassiveModule {
 
-// ApplyEvents applies a list of events to the AppModule.
-func (m *AppModule) ApplyEvents(eventsIn *events.EventList) (*events.EventList, error) {
-	// Events must be applied sequentially, since the application logic is not expected to be thread-safe.
-	return modules.ApplyEventsSequentially(eventsIn, m.ApplyEvent)
-}
-
-// ApplyEvent applies a single event to the AppModule.
-func (m *AppModule) ApplyEvent(event *eventpb.Event) (*events.EventList, error) {
-	switch e := event.Type.(type) {
-	case *eventpb.Event_Init:
-		// no actions on init
-	case *eventpb.Event_BatchFetcher:
-		switch e := e.BatchFetcher.Type.(type) {
-		case *bfpb.Event_NewOrderedBatch:
-			return m.applyNewOrderedBatch(e.NewOrderedBatch)
-		default:
-			return nil, fmt.Errorf("unexpected availability event type: %T", e)
-		}
-	case *eventpb.Event_App:
-		switch e := apppbtypes.EventFromPb(e.App).Type.(type) {
-		case *apppbtypes.Event_SnapshotRequest:
-			return m.applyAppSnapshotRequest(e.SnapshotRequest)
-		case *apppbtypes.Event_RestoreState:
-			return m.applyAppRestoreState(e.RestoreState)
-		case *apppbtypes.Event_NewEpoch:
-			return m.applyNewEpoch(e.NewEpoch)
-		default:
-			return nil, fmt.Errorf("unexpected app event type: %T", e)
-		}
-	case *eventpb.Event_Checkpoint:
-		switch e := checkpointpbtypes.EventFromPb(e.Checkpoint).Type.(type) {
-		case *checkpointpbtypes.Event_StableCheckpoint:
-			return m.applyStableCheckpoint((*checkpoint.StableCheckpoint)(e.StableCheckpoint))
-		default:
-			return nil, fmt.Errorf("unexpected checkpoint event type: %T", e)
-		}
-	default:
-		return nil, fmt.Errorf("unexpected type of App event: %T", event.Type)
+	appModule := &AppModule{
+		appLogic:  appLogic,
+		transport: transport,
 	}
 
-	return events.EmptyList(), nil
-}
+	m := dsl.NewModule("appModule")
 
-// The ImplementsModule method only serves the purpose of indicating that this is a Module and must not be called.
-func (m *AppModule) ImplementsModule() {}
+	// UponNewOrderedBatch sequentially apply a batch of ordered transactions to the application logic
+	// and returns an empty event list.
+	bfpbdsl.UponNewOrderedBatch(m, func(txs []*trantorpbtypes.Transaction) error {
 
-// applyNewOrderedBatch sequentially applies a batch of ordered transactions to the application logic
-// and returns an empty event list.
-func (m *AppModule) applyNewOrderedBatch(batch *bfpb.NewOrderedBatch) (*events.EventList, error) {
+		if err := appModule.appLogic.ApplyTXs(
+			sliceutil.Transform(
```

Comment:
Change the AppLogic interface to accept the generated type.

Reply:
done